app/build.gradle

```diff
@@ -52,14 +52,21 @@ android {
         timeOutInMs 10 * 60 * 1000  // 10 minutes
         installOptions "-d", "-t"
     }
+
+    productFlavors {
+      gms { }
+
+      fdroid { }
+    }
 }
 
 dependencies {
     compile fileTree(include: ['*.jar'], dir: 'libs')
     compile 'com.android.support:support-annotations:23.2.1'
     compile 'com.google.guava:guava:19.0'
-    compile 'com.google.android.gms:play-services-analytics:8.4.0'
-    compile 'com.google.android.gms:play-services-location:8.4.0'
+    gmsCompile 'com.google.android.gms:play-services-analytics:8.4.0'
+    gmsCompile 'com.google.android.gms:play-services-location:8.4.0'
+    fdroidCompile "com.android.support:support-v4:25.0.1"
```

Comment:
This should be `...:24.xxx` to match the `compileSdkVersion`.  Also, why do you need the support library for Fdroid but not the gms variant?